src/test/java/seedu/address/model/AddressBookTest.java

```diff
@@ -67,9 +68,14 @@ public void getTagList_modifyList_throwsUnsupportedOperationException() {
     public void getTodayReviewList() {
         AddressBook newData = getTypicalAddressBook();
         addressBook.resetData(newData);
-        List<Card> expectedList = addressBook.getCardList();
-        List<Card> actualList = addressBook.getTodayReviewList();
-        assertEquals(expectedList, actualList);
+        List<Card> originalList = addressBook.getCardList();
+        List<Card> reviewList = addressBook.getTodayReviewList();
+        assertEquals(originalList, reviewList);
+
+        // predicate filter should not include tomorrow's card
+        newData.getCardList().get(0).getSchedule().setRelativeNextReview(1);
+        List<Card> notEqualsList = addressBook.getTodayReviewList();
+        assertNotEquals(originalList, notEqualsList);
```

Comment:
this can fail for a variety of reasons, you should check specifically that the list does not include the card.